lib/bamboo/test.ex

```diff
@@ -90,23 +94,91 @@ defmodule Bamboo.Test do
       assert_delivered_email(unsent_email) # Will fail
   """
   def assert_delivered_email(%Bamboo.Email{} = email) do
-    import ExUnit.Assertions
     email = Bamboo.Mailer.normalize_addresses(email)
-    assert_received {:delivered_email, ^email}
+
+    do_assert_delivered_email(email)
+  end
+
+  defp do_assert_delivered_email(email) do
+    receive do
+      {:delivered_email, ^email} -> true
+    after
+      @timeout -> flunk_with_email_list(email)
+    end
+  end
+
+  defp flunk_with_email_list(email) do
+    if Enum.empty?(delivered_emails) do
+      flunk """
+      There were 0 emails delivered to this process.
+
+      If you expected an email to be sent, try these ideas:
+
+        1) Make sure you call deliver_now/1 or deliver_later/1 to deliver the email
+        2) Make sure you are using the Bamboo.TestAdapter
+        3) Use the process name feature of Bamboo.Test. This will allow Bamboo.Test
+           to work across processes: use Bamboo.Test, process_name: :my_test_name
+        4) If you are writing an acceptance test through a headless browser, see
```

Comment:
It does because if you are running acceptance test through a headless browser this is telling you to either write a controller test or use the process_name feature. 

I would like to guide people toward doing mostly controller tests and treating acceptance tests as super high level. So controller tests would check for sent emails, maybe check that some other stuff was done or saved to the db and acceptance tests check that it didn't fail and had something on the page.

I guess this is more of a personal preference, but I think this will lead to a faster/more maintainable test suite